block-lexical-variables/src/renderers/geras.js

```diff
@@ -0,0 +1,209 @@
+      const row = this.info_.rows[r];
+      if (row.getLastInput() instanceof IndentedInputMeasurable) {
+        this.drawIndentedInput_(row);
+      } else if (row.hasJaggedEdge) {
+        this.drawJaggedEdge_(row);
+      } else if (row.hasStatement) {
+        this.drawStatementInput_(row);
+      } else if (row.hasExternalInput) {
+        this.drawValueInput_(row);
+      } else {
+        this.drawRightSideRow_(row);
+      }
+    }
+    this.drawBottom_();
+    this.drawLeft_();
+  }
+
+  /**
+   * Draws an indented input in the block.
+   * @param {Blockly.blockRendering.Row} row the row being drawn
+   * @private
+   */
+  drawIndentedInput_(row) {
+    const input = row.getLastInput();
+    this.positionIndentedValueConnection_(row);
+    (/** @type {GerasHighlighter} */ this.highlighter_).drawIndentedInput(row);
+
+    const pathDown = (typeof input.shape.pathDown == 'function') ?
+      input.shape.pathDown(input.height) :
+      input.shape.pathDown;
+
+    this.outlinePath_ +=
+      Blockly.utils.svgPaths.lineOnAxis('H',
+          input.xPos + input.connectionWidth) +
+      Blockly.utils.svgPaths.lineOnAxis('v', input.connectionOffsetY) +
+      pathDown +
+      Blockly.utils.svgPaths.lineOnAxis('v',
+          row.height - input.connectionHeight - input.connectionOffsetY) +
+      Blockly.utils.svgPaths.lineOnAxis('H', input.xPos + input.width);
+  }
+
+  positionIndentedValueConnection_ = function(row) {
+    const input = row.getLastInput();
+    if (input.connectionModel) {
+      let connX = row.xPos + row.statementEdge + row.constants_.TAB_WIDTH;
+      if (this.info_.RTL) {
+        connX *= -1;
+      }
+      input.connectionModel.setOffsetInBlock(connX,
+          row.yPos + input.connectionOffsetY);
+    }
+  };
+}
+
+/**
+ * An extension of the Blockly Geras RenderInfo object that understands
+ * indented inputs.
+ */
+export class GerasRenderInfo extends Blockly.geras.RenderInfo {
+  /**
+   * Adds a measurable for the given input to the active row.
+   * @param {Blockly.Input} input the current input being processed
+   * @param {Blockly.blockRendering.Row} activeRow the active row
+   */
+  addInput_(input, activeRow) {
+    if (input instanceof IndentedInput) {
+      activeRow.elements.push(
+          new IndentedInputMeasurable(this.constants_, input)
+      );
+      activeRow.hasStatement = true;
+    } else {
+      super.addInput_(input, activeRow);
+    }
+  }
+
+  /**
+   * Determines whether a new row should be started given the next and last
+   * inputs.
+   * @param {Blockly.Input} input the input being considered
+   * @param {Blockly.Input} lastInput the previously processed input
+   * @returns {boolean} true if a new row should be started
+   */
+  shouldStartNewRow_(input, lastInput) {
+    console.log(input);
+    if (input instanceof IndentedInput) {
+      return true;
+    }
+    return super.shouldStartNewRow_(input, lastInput);
+  }
```

Comment:
I was looking at the source code for this this morning, and you may not need to override this any more now that you're subclassing statement input.

Reply:
This is correct. I've removed the call in the latest version.